lib/csv_file_from_public_host.rb

```diff
@@ -5,58 +5,50 @@ class FileEncodingError < ::EncodingError
 
   MAXIMUM_RANGE_BYTES = '300000'.freeze
 
-  def initialize(path)
-    @path = path
+  def self.csv_response(path, env: ENV)
+    connection = Faraday.new(url: Whitehall.public_root)
 
-    Tempfile.create(temp_fn, temp_dir, encoding: csv_file.encoding) do |tmp_file|
-      tmp_file.write(csv_file)
-      tmp_file.rewind
-      yield(tmp_file)
+    if env.has_key?("BASIC_AUTH_CREDENTIALS")
+      basic_auth_credentials = env["BASIC_AUTH_CREDENTIALS"].split(":")
+      basic_auth_user = basic_auth_credentials[0]
+      basic_auth_password = basic_auth_credentials[1]
+      connection.basic_auth(basic_auth_user, basic_auth_password)
     end
-  end
-
-private
 
-  def connection
-    conn = Faraday.new(url: Whitehall.public_root)
-    conn.basic_auth(basic_auth_user, basic_auth_password) if ENV.has_key?("BASIC_AUTH_CREDENTIALS")
-    conn
+    connection.get(path) do |req|
+      req.headers['Range'] = "bytes=0-#{MAXIMUM_RANGE_BYTES}"
+    end
   end
 
-  def response
-    connection.get(@path) do |req|
-      req.headers['Range'] = "bytes=0-#{MAXIMUM_RANGE_BYTES}"
+  def self.csv_preview_from(response)
```

Comment:
It's very minor but it'd be good to remove the unnecessary "self" from the title of the commit that extracts this method. The commit note is currently "Extract CsvFileFromPublicHost.self.csv_preview_from".

Reply:
Done.